src/generator.js

```diff
@@ -50,7 +58,7 @@ export const generateDescriptions = async (pages) => {
             return await generateDescription(page)
         } catch (error) {
             console.error(`Error generating description for ${page}: ${error}`);
-            return null;
+            return new DescriptionResult(page, null, GenerationStatus.UNKNOWN_ERROR);
```

Comment:
Instead of `UNKNOWN_ERROR` call this status `ERROR` and include details of the error in the `DescriptionResult` object.